packages/next/src/server/lib/route-resolver.ts

```diff
@@ -65,14 +76,52 @@ export async function makeResolver(dir: string, nextConfig: NextConfig) {
   const devServer = new DevServer({
     dir,
     conf: nextConfig,
+    hostname: 'localhost',
+    port: 3000,
   })
+
   await devServer.matchers.reload()
 
   // @ts-expect-error
   devServer.customRoutes = await loadCustomRoutes(nextConfig)
+  const matchers = getMiddlewareMatchers(middleware.matcher, nextConfig)
+  // @ts-expect-error
+  devServer.middleware = {
+    page: '/',
+    match: getMiddlewareRouteMatcher(matchers),
+    matchers,
+  }
+
+  type GetEdgeFunctionInfo =
+    typeof DevServer['prototype']['getEdgeFunctionInfo']
+  const getEdgeFunctionInfo = (
+    original: GetEdgeFunctionInfo
+  ): GetEdgeFunctionInfo => {
+    return (params: { page: string; middleware: boolean }) => {
+      if (params.middleware) {
+        return {
+          name: 'middleware',
+          paths: middleware.files.map((file) => join(process.cwd(), file)),
+          env: [],
+          wasm: [],
+          assets: [],
+        }
+      }
+      return original(params)
+    }
+  }
+  // @ts-expect-error protected
+  devServer.getEdgeFunctionInfo = getEdgeFunctionInfo(
+    // @ts-expect-error protected
+    devServer.getEdgeFunctionInfo.bind(devServer)
+  )
+  // @ts-expect-error protected
+  devServer.hasMiddleware = () => true
 
   const routeResults = new WeakMap<any, string>()
-  const routes = devServer.generateRoutes.bind(devServer)()
+  const routes = devServer.generateRoutes()
```

Comment:
Stylistic nits, the `.bind()()` doesn't accomplish anything and is kinda expensive.